geopandas/array.py

```diff
@@ -174,13 +230,18 @@ def from_wkb(data, crs=None):
     ----------
     data : array-like
         list or array of WKB objects
+        The 'srid' can be read from geoalchemy2's WKBElement.
     crs : value, optional
         Coordinate Reference System of the geometry objects. Can be anything accepted by
         :meth:`pyproj.CRS.from_user_input() <pyproj.crs.CRS.from_user_input>`,
         such as an authority string (eg "EPSG:4326") or a WKT string.
 
     """
-    return GeometryArray(vectorized.from_wkb(data), crs=crs)
+    converted_data = vectorized.from_wkb(data)
+    shared_crs = _get_common_crs_from_geometries(converted_data, crs)
+    if shared_crs is not None:
+        crs = shared_crs
+    return GeometryArray(converted_data, crs=crs)
```

Comment:
```suggestion
    geoms = vectorized.from_wkb(data)
    if not crs:
        crs = _get_common_crs_from_geometries(geoms, crs)
    return GeometryArray(geoms, crs=crs)
```

If `_get_common_crs_from_geometries ` returns None when nothing useful is found, we can simplify it like this.